rtlive/sources/data_be.py

```diff
@@ -1,39 +1,225 @@
+    'Antwerpen': 'ANT',
+    'Limburg': 'LIM',
+    'OostVlaanderen': 'EFL',
+    'VlaamsBrabant': 'FBR',
+    'WestVlaanderen': 'WFL',
+    'Hainaut': 'HAI',
+    'Liège': 'LIE',
+    'Luxembourg': 'LUX',
+    'Namur': 'NAM',
+    'BrabantWallon': 'WBR',   
+}
 
-def forecast_BE(df: pandas.DataFrame):
-    """ Applies testcount interpolation/extrapolation.
+BE_REGION_CODES = {
+    v : k
+    for k, v in BE_REGION_NAMES.items()
+}
 
-    Currently this assumes the OWID data, which only has an "all" region.
-    In the future, this should be replaced with more fine graned data loading!
+# Source: https://www.ibz.rrn.fgov.be/fileadmin/user_upload/fr/pop/statistiques/population-bevolking-20200101.pdf
+BE_REGION_POPULATION = {
+    'all':           11_476_279, # Belgium
+    'Flanders':       6_623_505,
+    'Wallonia':       3_641_748,
+    'Brussels':       1_211_026,
+    'Antwerpen':      1_867_366,
+    'Limburg':          876_785,
+    'OostVlaanderen': 1_524_077,
+    'VlaamsBrabant':  1_155_148,
+    'WestVlaanderen': 1_200_129,
+    'Hainaut':        1_345_270,
+    'Liège':          1_108_481,
+    'Luxembourg':       286_571,
+    'Namur':            495_474,
+    'BrabantWallon':    405_952
+}
+
+def get_data_BE(run_date) -> pandas.DataFrame:
+    """
+    Retrieve daily (run_date) regions and append national data (key 'all') to it
+    Parameters
+    ----------
+    run_date : pandas.Timestamp
+        date for which the data shall be downloaded
+    
+    Returns
+    -------
+    df : pandas.DataFrame
+        table with columns as required by rtlive/data.py API
+    """
+    
+    def redistribute(g, col):
+        gdata = g.groupby('REGION')[col].sum()
+        gdata.loc['Brussels'] += gdata.loc['Nan'] * (gdata.loc['Brussels']/(gdata.loc['Brussels'] + gdata.loc['Flanders'] + gdata.loc['Wallonia']))
+        gdata.loc['Flanders'] += gdata.loc['Nan'] * (gdata.loc['Flanders']/(gdata.loc['Brussels'] + gdata.loc['Flanders'] + gdata.loc['Wallonia']))
+        gdata.loc['Wallonia'] += gdata.loc['Nan'] * (gdata.loc['Wallonia']/(gdata.loc['Brussels'] + gdata.loc['Flanders'] + gdata.loc['Wallonia']))
+        gdata.drop(index='Nan', inplace=True)
+        gdata = numpy.round(gdata.fillna(0)).astype(int)
+        return gdata
+    
+    if run_date.date() > datetime.date.today():
+        raise ValueError('Run date is in the future. Nice try.')
+    if run_date.date() < datetime.date.today():
+        # TODO: implement downloading of historic data
+        raise NotImplementedError(
+            'Downloading with a run_date is not yet supported. '
+            f'Today: {datetime.date.today()}, run_date: {run_date}'
+        )
+        
+    # Download data from Sciensano and remove last two days to deal with reporting delays
+    df_tests = (pandas.read_csv('https://epistat.sciensano.be/Data/COVID19BE_tests.csv', parse_dates=['DATE'])
+       .set_index('DATE')
+       .sort_index()
+       .truncate(after=run_date - pandas.DateOffset(2))
```

Comment:
Depending on which reporting delay this truncation is about, the model may already account for it*!

*This is done through the "delay distribution" (see cell 9 of [this notebook](https://github.com/rtcovidlive/rtlive-global/blob/master/notebooks/Tutorial_model.ipynb)). Ideally, we'd need to calculate this empirical distribution differently for all countries.

Reply:
Nifty, I removed the truncate from the input timeline